src/server/services/procedures/salio/salio.js

```diff
@@ -0,0 +1,749 @@
+     * Only one radio button in each group may be checked.
+     * @param {string} sensor name of the sensor (matches at the end)
+     * @param {string} password current password for the sensor
+     * @param {BoundedNumber<0, 100>} x X position of the top left corner of the button (percentage).
+     * @param {BoundedNumber<0, 100>} y Y position of the top left corner of the button (percentage).
+     * @param {Number=} checkColor Color code of the check box itself (defaults to light blue).
+     * @param {Number=} textColor Color code of the text (if any) (defaults to black).
+     * @param {bool=} state Initial check state of the radio button (defaults to false) - All false in a group is OK, but you should avoid setting multiple explicitly to true.
+     * @param {String} id name of the radio button
+     * @param {String} group name of the group (only one radio button in each group can be selected by the user)
+     * @param {String} event name of the event to raise for click events
+     * @param {string} text The text to display next to the checkbox
+     * @returns {boolean} True if the action is successful, false otherwise.
+     */
+    SalIO.prototype.addRadioButton = function (sensor, password, x, y, checkColor=this.getColor(66, 135, 245), textColor=this.getColor(0, 0, 0), state=false, id, group, event, text) {
+        arguments[4] = checkColor;
+        arguments[5] = textColor;
+        return this._passToSensor('addRadioButton', arguments);
+    };
+    /**
+     * Add a custom label to the device.
+     * Labels are similar to buttons except they cannot be clicked.
+     * @param {string} sensor name of the sensor (matches at the end)
+     * @param {string} password current password for the sensor
+     * @param {BoundedNumber<0, 100>} x X position of the top left corner of the button (percentage).
+     * @param {BoundedNumber<0, 100>} y Y position of the top left corner of the button (percentage).
+     * @param {Number=} textColor Color code of the button text (if any) (defaults to black).
+     * @param {String} id Name of the label
+     * @param {string} text The text to display on the button
+     * @returns {boolean} True if the action is successful, false otherwise.
+     */
+    SalIO.prototype.addLabel = function (sensor, password, x, y, textColor=this.getColor(0, 0, 0), id, text) {
+        arguments[4] = textColor;
+        return this._passToSensor('addLabel', arguments);
+    };
+    /**
+     * Begin listening for GUI events such as button presses.
+     * @param {string} sensor name of the sensor (matches at the end)
+     * @param {string} password current password for the sensor
+     * @returns {boolean} True if the action is successful, false otherwise.
+     */
+    SalIO.prototype.guiListen = async function (sensor, password) {
+        const _sensor = await this._getSensor(sensor);
+        if (!_sensor) throw new Error('failed to connect to sensor');
```

Comment:
I would throw the error in `this._getSensor` unless there is a usage where we don't mind getting `undefined` for the sensor.